### ISSUE TITLE
Assigning to Hash slice with »=» no longer auto-vivifies the Hash

### ISSUE THREAD

**Issue by smls** [rakudo/rakudo#2718]
## The Regression

``` perl6
my $a;               # An uninitialized item container.
$a{1, 2, 3} »=» 42;  # Treat it as a Hash, take a slice, and hyper-assign to it.
dd $a;
```
Old (and expected) output: 

    Hash $a = ${"1" => 42, "2" => 42, "3" => 42}

New output:

    Any $a = Any

## Bisect

The commit that broke it: https://github.com/rakudo/rakudo/commit/b83179f0571b23eb3c656319a1d8bddad43c1a6a (Nov 2018)
Bisect log: https://gist.github.com/Whateverable/98b4c1aae9439945c57eac9d1886f883

## Variations

Same problem:

``` perl6
$a{1, 2, 3} X= 42;
```

``` perl6
$a{1, 2, 3} Z= 42, 42, 42;
```

But the following still work fine:

``` perl6
%a{1, 2, 3} »=» 42;  # Using an existing Hash object (i.e. no need to auto-vivify the Hash)
```

``` perl6
$a{1, 2, 3} = 42, 42, 42;  # Using normal list assignment
```

## Environment

This is Rakudo version 2018.12-312-gf98a70cab built on MoarVM version 2018.12-114-g7ca628263
implementing Perl 6.d.

**Comment by AlexDaniel** [comment]
Obligatory ping: @lizmat :)

**Comment by lizmat** [comment]
Fixed with 616b757a46 , tests added with https://github.com/perl6/roast/commit/a3b365fcda
